Sources/Account/Account+SCIOperations.swift

```diff
@@ -0,0 +1,169 @@
+                fogMerkleProofService: fogMerkleProofService,
+                fogResolverManager: fogResolverManager,
+                mixinSelectionStrategy: mixinSelectionStrategy,
+                rngSeed: rngSeed,
+                targetQueue: targetQueue)
+        }
+
+        private func verifyBlockVersion(
+            _ blockVersion: BlockVersion,
+            _ completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>) -> Void
+        ) -> Bool {
+            // verify block version >= 3
+            guard blockVersion >= 3 else {
+                serialQueue.async {
+                    completion(.failure(.requiresBlockVersion3(
+                        "Block version must be > 3 for SCI support")))
+                }
+                return false
+            }
+            return true
+        }
+
+        private func verifyAmountIsNonZero(
+            _ amount: Amount,
+            _ actionDescription: String,
+            _ completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>) -> Void
+        ) -> Bool {
+            guard amount.value > 0 else {
+                let errorMessage = "createSignedContingentInput failure: " +
+                    "Cannot \(actionDescription) 0 \(amount.tokenId)"
+                logger.error(errorMessage, logFunction: false)
+                serialQueue.async {
+                    completion(.failure(.invalidInput(errorMessage)))
+                }
+                return false
+            }
+            return true
+        }
+
+        private func logTxOuts(_ txOuts: [KnownTxOut], _ message: String) {
+            logger.info(
+                "\(message): " +
+                    """
+                        0x\(redacting: txOuts.map {
+                            $0.publicKey.hexEncodedString()
+                        })
+                    """,
+                logFunction: false)
+        }
+
+        private func logUnspentTxOuts(
+            _ recipient: PublicAddress,
+            _ amountToSend: Amount,
+            _ amountToReceive: Amount,
+            _ txOuts: [KnownTxOut]
+        ) {
+            logger.info(
+                "Creating signed contingent input to recipient: \(redacting: recipient), " +
+                    "amountToSend: \(redacting: amountToSend), " +
+                    "amountToRecieve: \(redacting: amountToReceive), " +
+                    "unspentTxOutValues: \(redacting: txOuts.map { $0.value })",
+                logFunction: false)
+        }
+
+        func createSignedContingentInput(
+            to recipient: PublicAddress,
+            memoType: MemoType,
+            amountToSend: Amount,
+            amountToReceive: Amount,
+            completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>
+            ) -> Void
+        ) {
+            let functionName = "createSignedContingentInput"
```

Comment:
this can be done automatically by the logger - https://github.com/mobilecoinofficial/MobileCoin-Swift/blob/master/Sources/Common/MobileCoinLogging.swift

Reply:
awesome - tx

Reply:
Ah, ya, I don't recall for sure, but I explicitly disabled the logging of the function name in the default manner to be able to log "<function name> failure" in the log... But sure, no need to do that - can use the default.  Switched it out.